RELEASE.md

```diff
@@ -87,12 +69,48 @@ the same notes will be used in the github release.
     git add CHANGELOG.md
     git commit -s -m "Modify changelog for $MAJOR.$MINOR.$PATCH release"
 
+## Modify the version constant in the Makefile to match the new release
+
+Usually this only consists of dropping the `-dev` suffix from the string.
+
+    VERSION="$MAJOR.$MINOR.$PATCH"
+
+Commit and push the changes to the prep branch
+
+    git add Makefile
+    git commit -s -m "Modify version to $MAJOR.$MINOR.$PATCH"
+
 ## Push the prep branch and open a Pull Request
 
 The pull request has to be `v$MAJOR.$MINOR.$PATCH-prep -> v$MAJOR.$MINOR`
 
 Once the pull request is approved and merged, a tag can be created.
 
+## Modify the version constant on the master branch, if needed
+
+After branching out from the tree for release, the version need to be updated
+to reflect the next planned release, i.e.
+
+    VERSION="$MAJOR.<$MINOR+1>.0-dev"
+
+## Update the changelog in the master branch
+
+Once the release PR has been merged, the changelog in the master branch needs to
+be updated. If the release is the latest stable version, this can be done by
+simply copying the file over from the stable branch. For example, from the
+`master` branch, one can run the following command (assuming v0.7 is the latest
+stable release branch):
```

Comment:
> If the release is the latest stable version, this can be done by
simply copying the file over from the stable branch.

That's not necessarily correct. `0.7.1` can be released after `0.6.X` and will be missing changes from `0.6.*`. It might be best to copy over the changelog section into the existing changelog file on master.